src/chart/context.rs

```diff
@@ -268,18 +268,25 @@ impl<'a, DB: DrawingBackend, X: Ranged, Y: Ranged> ChartContext<'a, DB, RangedCo
         }
 
         if let Some(style) = axis_style {
-            let mut x0 = if orientation.0 > 0 { 0 } else { tw as i32 };
-            let mut y0 = if orientation.1 > 0 { 0 } else { th as i32 };
+            let mut x0 = if orientation.0 > 0 { 0 }  else { tw as i32 };
+            let mut y0 = if orientation.1 > 0 { 0 }  else { th as i32 };
             let mut x1 = if orientation.0 >= 0 { 0 } else { tw as i32 };
             let mut y1 = if orientation.1 >= 0 { 0 } else { th as i32 };
 
+
             if orientation.0 == 0 {
                 x0 = axis_range.start;
                 x1 = axis_range.end;
             } else {
                 y0 = axis_range.start;
                 y1 = axis_range.end;
             }
+
+            if area.is_inset() {
```

Comment:
Just little bit worry about the logic coupling. Since the drawing area is originally designed not to include too many details about chart. 
And later we may have different types of weird coordinate system, and so I am prefer this flag lives in the 2D chart context rather than the generic drawing area.

Reply:
Since my design needs to be implemented at builder stage it seemed logical to set inset field in `Area `structure. It looks good for me. Pretty generic flag :)